script.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+// form issue 해결
+// submit이벤트가 발생하면 브라우저가refresh된다.
+// e.preventDefault()를 통해서 브라우저의 refresh를 막고
+// 새로운 변수 location에 input값 저장해서 textContent로 출력
+const Form = document.querySelector("form");
+const inputBox = document.querySelector("input");
+const waitingList = document.querySelector("#waitingList");
+
+Form.addEventListener("submit", (e) => {
+  e.preventDefault();
+  //const location = inputBox.value;
+  //waitingList.textContent = location;
+});
+
+// input창에 입력
+let itemList = [];
+let finishedItem = [];
+
+let inputButton = document.querySelector("button");
+inputButton.addEventListener("click", addItem);
+
+function addItem() {
+  let item = document.querySelector("#input").value;
+  if (item != null) {
+    itemList.push(item);
+    document.querySelector("#input").value = "";
+    document.querySelector("#input").focus();
+  }
+  showList();
+}
+
+// ul출력
+
+function showList() {
+  let list = "";
+  for (let i = 0; i < itemList.length; i++) {
+    list +=
+      "<li>" +
+      itemList[i] +
+      "<button class='close' id=" +
+      i +
+      ">" +
+      "<img src='img/bin.png'/>" +
+      "</button></li>";
+    // list +=
+    //   "<li id='list" +
+    //   i +
+    //   "'><button class='listBtn' onclick='movingFinish(" +
+    //   i +
+    //   ")'>" +
+    //   "<span class='text'>" +
+    //   itemList[i] +
+    //   "</span></button><button class='close' id=" +
+    //   i +
+    //   ">" +
+    //   "<img src='img/bin.png'/>" +
+    //   "</button></li>";
+  }
+  document.getElementById("waitingList").innerHTML = list;
+  console.log(itemList);
+  waitingCount();
+
+  let deleteButtons = document.querySelectorAll(".close");
+  for (let i = 0; i, deleteButtons.length; i++) {
+    deleteButtons[i].addEventListener("click", deleteItem);
+  }
```

Comment:
js의 array method를 사용하면 코드가 훨씬 간결해진답니다!
```suggestion
  deleteButtons.forEach((btn) => {
    btn.addEventListener("click", deleteItem);
  })
```

Reply:
이 부분은 존재하는 모든 삭제 버튼 요소를 불러와서 각각에 이벤트 리스너를 등록한 코드같은데, **이벤트 위임** 이락고 이벤트 캡쳐링과 버블링이라는 DOM의 특징을 활용해보면 좋을 것 같아요..!
(delete buttons의 부모 요소인 li에 이벤트 리스너를 한 번 걸어서 삭제 버튼을 눌렀는지 event.target 을 활용해서 분기를 해주면 좋을 것 같아요!)

Reply:
이 부분은 존재하는 모든 삭제 버튼 요소를 불러와서 각각에 이벤트 리스너를 등록한 코드같은데, **이벤트 위임** 이락고 이벤트 캡쳐링과 버블링이라는 DOM의 특징을 활용해보면 좋을 것 같아요..!
[event delegation](https://developer.mozilla.org/ko/docs/Learn/JavaScript/Building_blocks/Events)
(delete buttons의 부모 요소인 li에 이벤트 리스너를 한 번 걸어서 삭제 버튼을 눌렀는지 event.target 을 활용해서 분기를 해주면 좋을 것 같아요!)